### PR TITLE
fix missing error handling when creating mutatingwebhookconfiguration

### PR DESCRIPTION
**Pull request by akumor**
Signed-off-by: akumor <akumor@users.noreply.github.com>

# TL;DR
This PR adds error handling that makes the flyte-pod-webhook fail correctly when it is unable to create the mutatingwebhookconfiguration instead of silently ignoring the error.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
To ensure that the flyte-pod-webhook fails in an intuitive way, this PR adds a log line and returns an error instead of nil from the function responsible for creating the MutatingWebhookConfiguration.

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/2167

### CODE REVIEW

**Review by EngHabu — APPROVED** [review]
👍🏽
